### PR TITLE
Reduce Chance Of Flakiness By Submitting Metrics More Often

### PR DESCRIPTION
**Pull request by sethAmazon**
# Description of the issue
Flaky test

# Description of changes
Add more metrics per minute to reduce the chance of not publishing or indexing in cw

# License
By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

# Tests
export AWS_REGION=us-west-2
go test -computeType=EC2 -p 1 -v
from within the dir on ec2 instance

### CODE REVIEW

File: test/metrics_number_dimension/resources/10_dimension/config.json [hunk 1]
```diff
@@ -1,11 +1,12 @@
 {
   "agent": {
-    "metrics_collection_interval": 60,
+    "metrics_collection_interval": 10,
```
Comment (sethAmazon) [inline test/metrics_number_dimension/resources/10_dimension/config.json:3]: Send to backend faster and more often, so less chance of timing error.

File: test/metrics_number_dimension/metrics_number_dimension_test.go [hunk 2]
```diff
@@ -96,7 +96,7 @@ func TestNumberMetricDimension(t *testing.T) {
 
 			// test for cloud watch metrics
 			dimensionFilter := buildDimensionFilterList(parameter.numberDimensionsInCW)
-			awsservice.ValidateMetricWithTest(t, parameter.metricName, namespace, dimensionFilter)
+			awsservice.ValidateMetricWithTest(t, parameter.metricName, namespace, dimensionFilter, 5, 20*time.Second)
```
Comment (sethAmazon) [inline test/metrics_number_dimension/metrics_number_dimension_test.go:99]: Increase time to try to get metric so less chance of metric not being indexed.

File: test/metrics_number_dimension/resources/10_dimension/config.json (hunk 1 above)
Comment (sethAmazon) [reply test/metrics_number_dimension/resources/10_dimension/config.json:3]: Furthermore, my understanding is the more metrics you have the faster cw will index.

**Review by okankoAMZ — APPROVED** [review]

File: util/awsservice/cloudwatchmetrics.go [hunk 3]
```diff
@@ -57,10 +57,18 @@ func ValidateMetric(metricName, namespace string, dimensionsFilter []types.Dimen
 }
 
 // ValidateMetrics takes the metric name, metric dimension and corresponding namespace that contains the metric
-func ValidateMetricWithTest(t *testing.T, metricName, namespace string, dimensionsFilter []types.DimensionFilter) {
-	err := ValidateMetric(metricName, namespace, dimensionsFilter)
+func ValidateMetricWithTest(t *testing.T, metricName, namespace string, dimensionsFilter []types.DimensionFilter, retries int, retryTime time.Duration) {
+	var err error
+	for i := 0; i < retries; i++ {
+		err = ValidateMetric(metricName, namespace, dimensionsFilter)
```
Comment (adam-mateen) [inline util/awsservice/cloudwatchmetrics.go:63]: nit: You could log that retry `i` variable so it is a bit clearer in the logs that retries are occurring.
Or maybe it is already  sufficiently clear in the logs.

**Review by adam-mateen — APPROVED** [review]

File: util/awsservice/cloudwatchmetrics.go [hunk 4]
```diff
@@ -57,10 +57,18 @@ func ValidateMetric(metricName, namespace string, dimensionsFilter []types.Dimen
...
+		if err == nil {
+			return
+		}
+		log.Printf("could not validate metrics try : %d of %d error %v", i+1, retries, err)
```
Comment (jefchien) [inline util/awsservice/cloudwatchmetrics.go:67]: Can we log the retry time? Something similar to the fargate tests. 

> 2023/07/27 21:19:58 Current retry: 3/15 and begin to sleep for 20s

**Pull request merged**